### ISSUE TITLE
Null Reference When Drawing an SVG USE Element

### ISSUE THREAD

**Issue by rfoltz** [vvvv/SVG#74]
Alright my issue is when there is a SvgUse element inside of an SvgDocument and you try to draw a Bitmap you get a null reference exception in the following method in SvgElement.cs This is very similar to issue #64 in this repo.

```
protected virtual void RenderChildren(SvgRenderer renderer)
{
    foreach (SvgElement element in this.Children)
    {
         element.Render(renderer);
     }
 }
```

Here's some example code below:

```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0" y="0" width="768" height="1024">
    <defs>
        <rect x="22.85714" y="62.36218" width="464.2857" height="274.2857" fill="#C8B7B7" id="pieChartRect" />
    </defs>
    <use x="250.7143" y="406.6479" id="usePieChart1" xlink:href="#pieChartRect"/>
</svg>
```

Now the C# code:

```
SvgDocument newSVGDoc = SvgDocument.Open("C:\svgDoc.svg");
//get the use element and re-position it.
SvgUse usePieChart1 = newSVGDoc.GetElementById<SvgUse>("usePieChart1");
usePieChart1.X = 40;
usePieChart1.Y = 60;
```

Some interesting facts are that even though the document has the `xmlns:xlink="http://www.w3.org/1999/xlink"` namespace on the svg element the SvgDocument object does not have it in it's CustomAttributes for some reason.

So can someone with more experience with this code base could take a look at the issue. Maybe @tebjan ?

Thanks!

**Comment by jamescauwelier** [comment]
I have a similar issue as well:

`
NullReferenceException: Object reference not set to an instance of an object
Svg.SvgElementIdManager.GetElementById (System.Uri uri) (at C:/Dev/vvvv/SVG/Source/SvgElementIdManager.cs:44)
Svg.SvgUse.Render (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/Document Structure/SvgUse.cs:77)
Svg.SvgElement.RenderChildren (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/SvgElement.cs:726)
Svg.SvgVisualElement.Render (ISvgRenderer renderer, Boolean renderFilter) (at C:/Dev/vvvv/SVG/Source/Basic Shapes/SvgVisualElement.cs:183)
Svg.SvgVisualElement.Render (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/Basic Shapes/SvgVisualElement.cs:129)
Svg.SvgElement.RenderChildren (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/SvgElement.cs:726)
Svg.SvgVisualElement.Render (ISvgRenderer renderer, Boolean renderFilter) (at C:/Dev/vvvv/SVG/Source/Basic Shapes/SvgVisualElement.cs:183)
Svg.SvgVisualElement.Render (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/Basic Shapes/SvgVisualElement.cs:129)
Svg.SvgElement.RenderChildren (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/SvgElement.cs:726)
Svg.SvgElement.Render (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/SvgElement.cs:714)
Svg.SvgFragment.Render (ISvgRenderer renderer) (at C:/Dev/vvvv/SVG/Source/Document Structure/SvgFragment.cs:169)
Svg.SvgDocument.Draw (System.Drawing.Bitmap bitmap) (at C:/Dev/vvvv/SVG/Source/SvgDocument.cs:496)
`

The SVG I am trying to draw is:

`
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg x="0" y="0" width="40px" height="1000px" overflow="scroll" viewBox="0, 0, 128, 3200" preserveAspectRatio="xMinYMin" font-size="0" xml:space="default" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xml="http://www.w3.org/XML/1998/namespace" version="1.1">
  <g transform="translate(0, 1200)" xml:space="default">
    <g xml:space="default">
      <defs xml:space="default">
        <g id="E000" xml:space="default">
          <g transform="scale(0.4, -0.4)" xml:space="default">
            <path d="M20 498 Q43 484 62.5 438.5 Q82 393 82 350 Q82 339 81 334 Q78 311 61 234.5 Q44 158 44 128 Q44 94 54.5 61.5 Q65 29 72 17 Q77 11 77 6 Q77 4 75 2 Q73 0 71 0 Q68 0 63 6 Q14 68 14 192 Q14 229 31.5 295.5 Q49 362 49 397 Q49 416 46 430.5 Q43 445 37 455.5 Q31 466 26.5 472 Q22 478 13 486.5 Q4 495 2 498 Q26 526 37.5 548.5 Q49 571 49 603 Q49 630 31.5 690.5 Q14 751 14 805 Q14 930 63 990 Q69 997 71 997 Q72 997 75 994 Q77 994 77 990 Q77 985 72 980 Q44 930 44 869 Q44 839 61 762 Q78 685 81 662 Q82 657 82 646 Q82 603 62.5 557.5 Q43 512 20 498 z" xml:space="default" horiz-adv-x="84" />
          </g>
        </g>
      </defs>
      <use xlink:href="#E000" x="0" y="300" transform="scale(4)" xml:space="default" class="brace" height="400" width="400" />
    </g>
  </g>
</svg>
`

**Comment by jamescauwelier** [comment]
Nevermind... Just discovered that System.Drawing is not available in Unity and I need to find an alternative way to render in Png or use a library that can visualize SVG in Unity.  The latter approach gave me low quality results though, but we'll see what happens.

**Comment by mrbean-bremen** [comment]
Looks like this is the same problem that will be fixed with #377.
